src/gravel/controllers/nodes/mgr.py

```diff
@@ -412,39 +412,61 @@ async def _spawn_etcd(
         hostname = self._state.hostname
         address = self._state.address
 
-        logger.info(f"starting etcd, hostname: {hostname}, addr: {address}")
-
-        client_url: str = f"http://{address}:2379"
-        peer_url: str = f"http://{address}:2380"
-
-        if not initial_cluster:
-            initial_cluster = f"{hostname}={peer_url}"
-
-        args_dict: Dict[str, str] = {
-            "name": hostname,
-            "initial-advertise-peer-urls": peer_url,
-            "listen-peer-urls": peer_url,
-            "listen-client-urls": f"{client_url},http://127.0.0.1:2379",
-            "advertise-client-urls": client_url,
-            "initial-cluster": initial_cluster,
-            "initial-cluster-state": "existing",
-            "data-dir": f"/var/lib/etcd/{hostname}.etcd"
-        }
+        data_dir: Path = Path("/var/lib/etcd")
+        if not data_dir.exists():
+            data_dir.mkdir(0o700)
 
-        if new:
-            assert token
-            args_dict["initial-cluster-token"] = token
-            args_dict["initial-cluster-state"] = "new"
-
-        args = " ".join([f"--{k} {v}" for k, v in args_dict.items()])
-        logger.debug(f"spawn etcd: {args}")
-        etcd_cmd = f"etcd {args}"
+        logger.info(f"starting etcd, hostname: {hostname}, addr: {address}")
 
+        def _get_etcd_args() -> List[str]:
+            client_url: str = f"http://{address}:2379"
+            peer_url: str = f"http://{address}:2380"
+
+            nonlocal initial_cluster
+            if not initial_cluster:
+                initial_cluster = f"{hostname}={peer_url}"
+
+            args: List[str] = [
+                "--name", hostname,
+                "--initial-advertise-peer-urls", peer_url,
+                "--listen-peer-urls", peer_url,
+                "--listen-client-urls", f"{client_url},http://127.0.0.1:2379",
+                "--advertise-client-urls", client_url,
+                "--initial-cluster", initial_cluster,
+                "--initial-cluster-state", "existing",
+                "--data-dir", str(data_dir),
+            ]
+
+            if new:
+                assert token
+                args += ["--initial-cluster-state", "new"]
+                args += ["--initial-cluster-token", token]
+
+            return args
+
+        def _get_container_cmd():
+            registry = "quay.io/coreos/etcd"
```

Comment:
this should be a config option imo

Reply:
etcd conf options added for `registry`, `version`, and `data_dir`.